contracts/cw-proposal-single/src/contract.rs

```diff
@@ -711,9 +713,90 @@ pub fn query_info(deps: Deps) -> StdResult<Binary> {
 }
 
 #[cfg_attr(not(feature = "library"), entry_point)]
-pub fn migrate(_deps: DepsMut, _env: Env, _msg: MigrateMsg) -> Result<Response, ContractError> {
-    // Don't do any state migrations.
-    Ok(Response::default())
+pub fn migrate(deps: DepsMut, env: Env, msg: MigrateMsg) -> Result<Response, ContractError> {
+    #[derive(Serialize, Deserialize, Clone, PartialEq, JsonSchema, Debug)]
+    struct BetaProposal {
+        pub title: String,
+        pub description: String,
+        /// The address that created this proposal.
+        pub proposer: Addr,
+        /// The block height at which this proposal was created. Voting
+        /// power queries should query for voting power at this block
+        /// height.
+        pub start_height: u64,
+        /// The minimum amount of time this proposal must remain open for
+        /// voting. The proposal may not pass unless this is expired or
+        /// None.
+        pub min_voting_period: Option<Expiration>,
+        /// The the time at which this proposal will expire and close for
+        /// additional votes.
+        pub expiration: Expiration,
+        /// The threshold at which this proposal will pass.
+        pub threshold: Threshold,
+        /// The total amount of voting power at the time of this
+        /// proposal's creation.
+        pub total_power: Uint128,
+        /// The messages that will be executed should this proposal pass.
+        pub msgs: Vec<CosmosMsg<Empty>>,
+
+        pub status: Status,
+        pub votes: Votes,
+        pub allow_revoting: bool,
+
+        /// Information about the deposit that was sent as part of this
+        /// proposal. None if no deposit.
+        pub deposit_info: Option<CheckedDepositInfo>,
+    }
+
+    match msg {
+        MigrateMsg::FromBeta {} => {
+            // Retrieve current map from storage
+            let current_map: Map<u64, BetaProposal> = Map::new("proposals");
+            let count = current_map
+                .keys(deps.storage, None, None, Order::Ascending)
+                .count();
+            println!("keys count {}", count);
+
+            let current = current_map
+                .range(deps.storage, None, None, Order::Ascending)
+                .collect::<StdResult<Vec<(u64, BetaProposal)>>>()?;
+
+            // Add migrated entries to new map.
+            // TODO: determine whether this needs to be paginated
```

Comment:
pending gas analysis

Reply:
We can profile the gas usage with varying numbers of proposals here. It would be really useful if we could just find the max and average number of proposals in mainnet so we can use real numbers. Not sure if that's easy to get from the indexer or somewhere.

Reply:
Here are the top 5 proposal counts for daos in mainnet right now (thanks for the raw data @ezekiiel):
```
5
5
5
7
16
```

So, we can just test the gas usage for 20 proposals, to know the worst case scenario here.

Reply:
Also, the 16 is 🥇 🐾 dog DAO 🐾  🥇

Reply:
update, I was able to migrate at least 220 proposals at a time without running out of gas (unable to determine true limit because I ran out of junox and my local docker container keeps crashing) but I think we are good to proceed without pagination for now